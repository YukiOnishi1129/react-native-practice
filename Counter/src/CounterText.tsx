import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 20,
  },
});

interface Props {
  count: number;
}

export default function CounterText(props: Props) {
  const {count} = props;
  return (
    <View style={styles.container}>
      {count >= 10 && <Text>you're great!</Text>}
      <Text style={styles.text}>{count}</Text>
    </View>
  );
}
