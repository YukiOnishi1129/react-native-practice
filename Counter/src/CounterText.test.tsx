import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CounterText from './CounterText';

describe('CounterText', () => {
  it('通常のCounterTextを描画する', () => {
    const component = renderer.create(<CounterText count={0} />);
    // toMatchSnapshot: JSXのスナップショットを書き出す
    expect(component).toMatchSnapshot();
  });
  it('カウントが10以上の時のテキストが表示されているCounterTextを描画できる', () => {
    const component = renderer.create(<CounterText count={10} />);
    expect(component).toMatchSnapshot();
  });
});
