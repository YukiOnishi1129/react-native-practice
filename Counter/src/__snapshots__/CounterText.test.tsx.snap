// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterText カウントが10以上の時のテキストが表示されているCounterTextを描画できる 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text>
    you're great!
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    10
  </Text>
</View>
`;

exports[`CounterText 通常のCounterTextを描画する 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    0
  </Text>
</View>
`;
